### PR TITLE
155 - Person-specific calendar integrations

### PR DESCRIPTION
**Pull request by amiedes**
Closes [#155](https://github.com/PopulateTools/issues/issues/155)

### What does this PR do?

* Deattaches calendar integration configurations from sites so users can chose their own calendar service.
* Makes some work to prepare for when we permit syncing processes events from calendars in a similar way.

pr-calendar

### How should this be manually tested?

Go to a staging site, configure different calendar integrations for two people in that site, and manually run agendas synchronization.

### Does this PR changes any configuration file?

No

### Future work

* I renamed most of the relevant models, methods, variables etc. A few well-isolated components (`lib/services/gobierto_people/*`) still use names which suggest calendar integrations are only related to people.

### CODE REVIEW

**Comment by amiedes** [comment]
I just realised I need to apply the permissions logic. I'll do this  now.

File: app/controllers/gobierto_people/people/google_calendar/authorization_controller.rb [hunk 1]
```diff
@@ -42,7 +42,8 @@ def load_person
         end
 
         def load_google_calendar_configuration
-          @configuration = ::GobiertoPeople::PersonGoogleCalendarConfiguration.find_or_initialize_by person_id: @person.id
+          @configuration = ::GobiertoCalendars::GoogleCalendarConfiguration.find_by(collection_id: @person.calendar.id)
```
Comment (ferblape) [inline app/controllers/gobierto_people/people/google_calendar/authorization_controller.rb]: This configuration might not exist. See this Rollbar: https://rollbar.com/Populate/gobierto/items/646/

File: config/locales/gobierto_admin/gobierto_calendars/models/ca.yml [hunk 2]
```diff
@@ -2,6 +2,13 @@
 ca:
   activemodel:
     attributes:
+      gobierto_admin/gobierto_calendars/calendar_configuration_form:
+        ibm_notes_pwd: La contrasenya
+        ibm_notes_url: La URL
+        ibm_notes_usr: La compte
```
Comment (ferblape) [inline config/locales/gobierto_admin/gobierto_calendars/models/ca.yml]: El nom d'usuari

File: config/locales/gobierto_admin/gobierto_calendars/models/ca.yml [hunk 3]
```diff
@@ -2,6 +2,13 @@
...
+        microsoft_exchange_pwd: La contrasenya
+        microsoft_exchange_url: La URL
+        microsoft_exchange_usr: La compte
```
Comment (ferblape) [inline config/locales/gobierto_admin/gobierto_calendars/models/ca.yml]: El nom d'usuari

File: db/data/20171016104939_encrypt_calendar_configuration_credentials.rb [hunk 4]
```diff
@@ -1,35 +1,37 @@
...
 
   def self.up
-    Site.with_agendas_integration_enabled.each do |site|
-      gp_module_settings = site.gobierto_people_settings
+    puts "\n\t[DATA MIGRATION SKIPPED DUE TO INCOMPATIBILITIES]\n\n"
+    # Site.with_agendas_integration_enabled.each do |site|
+    #   gp_module_settings = site.gobierto_people_settings
 
-      next if gp_module_settings.calendar_integration != 'ibm_notes'
+    #   next if gp_module_settings.calendar_integration != 'ibm_notes'
```
Comment (ferblape) [inline db/data/20171016104939_encrypt_calendar_configuration_credentials.rb]: Let's remove these comments

File: app/forms/gobierto_admin/gobierto_calendars/calendar_configuration_form.rb [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+      def ibm_notes_configuration_class
+        ::GobiertoCalendars::IbmNotesCalendarConfiguration
+      end
+
+      def microsoft_exchange_configuration_class
+        ::GobiertoCalendars::MicrosoftExchangeCalendarConfiguration
+      end
+
+      def collection
+        @collection ||= ::GobiertoCommon::Collection.find_by(id: collection_id)
```
Comment (ferblape) [inline app/forms/gobierto_admin/gobierto_calendars/calendar_configuration_form.rb]: Filter by site

File: app/controllers/gobierto_admin/gobierto_calendars/calendar_configuration_controller.rb [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        else
+          load_calendar_integrations
+          render 'gobierto_admin/gobierto_calendars/calendar_configuration/edit'
+        end
+      end
+
+      private
+
+      def load_collection
+        @collection = ::GobiertoCommon::Collection.find(params[:id])
```
Comment (ferblape) [inline app/controllers/gobierto_admin/gobierto_calendars/calendar_configuration_controller.rb]: Filter by site

**Review by ferblape — CHANGES_REQUESTED** [review]
Great work! There are a few minor detals and the Rollbar I have mentioned. To reproduce it follow these steps:

1. Configure an IBM integration
2. Import events
3. Remove configuration
4. Configure a Google Calendar integration
5. 💥

**Comment by amiedes** [comment]
Fixed 👷

**Pull request merged**